shadow_cache.c

```diff
@@ -95,11 +96,11 @@ void shadow_cache_put(const char *name, const char *entry TSRMLS_DC)
 	}
 	namelen = shadow_cache_segmented_name(&segname, name TSRMLS_CC);
 	zend_string *segname_zs = zend_string_init(segname, namelen, 1);
-	entry_zs = zend_string_init(entry, strlen(entry), 0);
+	entry_zs = zend_string_init(entry, strlen(entry), 1);
 	ZVAL_STR(&entry_zv, entry_zs);
 	zend_hash_update(&SHADOW_G(cache), segname_zs, &entry_zv);
 	efree(segname);
-    zend_string_release(segname_zs);
+	zend_string_release(segname_zs);
```

Comment:
We shouldn't free `entry_zs`, as it's stored in hash table. It will be freed when we destroy hash table.